core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java

```diff
@@ -296,7 +296,20 @@ public static InetSocketAddress getBindAddress(ServiceType service) {
   }
 
   /**
-   * Gets a local host name for the host this JVM is running on.
+   * Gets the local hostname to be used by the client. If this isn't configured, a non-loopback
+   * local hostname will be looked up.
+   *
+   * @return the local hostname for the client
+   */
+  public static String getClientHostName() {
+    if (Configuration.containsKey(PropertyKey.USER_HOSTNAME)) {
+      return Configuration.get(PropertyKey.USER_HOSTNAME);
+    }
+    return getLocalHostName();
```

Comment:
I guess what I am curious about is, what still uses `getLocalHostName()` directly?

If nothing should directly call `getLocalHostName()`, then it should be changed to private. Also, I am assuming that no binding type of call will ever call `getClientHostName()` or `getLocalHostName()`?

If some sites still call `getLocalHostName()`, what is the distinction? Which sites should use the new method and which ones should use `getLocalHostName()`?